oauth2-api/src/main/java/net/oauth2/AccessToken.java

```diff
@@ -77,11 +78,18 @@ public AccessToken(Map<String, Object> map) {
 			throw new IllegalArgumentException("map is null");
 		this.accessToken = (String) map.get("access_token");
 		this.tokenType = (String) map.getOrDefault("token_type", "Bearer");
-		Long val = (Long) map.get("expires_in");
-		if(val!=null)
-			this.expiresIn = Long.parseLong(String.valueOf(val));
+		if(map.containsKey("expires_in")){
+			//Create a new long so that the case where expires_in is an integer is handled.
+			this.expiresIn = new Long(String.valueOf(map.get("expires_in")));
+		}
```

Comment:
I was getting errors when the expires in value was small enough to be an integer value.